### ISSUE TITLE
[Feature request]: Custom filter doesn't apply when using tabpages

### ISSUE THREAD

**Issue by GabMus** [akinsho/bufferline.nvim#398]
### What happened?

I have set bufferline to tabs mode, and I expected the `custom_filter` callback to apply to this mode as well, but turns out it doesn't.

Here's my bufferline config:

```lua
require('bufferline').setup{
    options = {
        mode = 'tabs',  -- only show native tabs, alternative: 'buffers'
        numbers = 'none',
        diagnostics = 'nvim_lsp',
        always_show_bufferline = true,
        separator_style = 'thin',  -- slant, thick
        custom_filter = function(buf_number, buf_numbers)
            -- filter out filetypes you don't want to see
            if
                vim.bo[buf_number].filetype ~= "packer" and
                vim.bo[buf_number].filetype ~= "terminal" and
                vim.bo[buf_number].filetype ~= "TelescopePrompt" and
                vim.bo[buf_number].filetype ~= "alpha" and
                vim.bo[buf_number].filetype ~= "NvimTree"
            then
                return true
            else
                return false
            end
        end,
    }
}
```

### What did you expect to happen?

I'd like the `custom_filter` to be able to filter out tab names. For instance, I usually open new tabs by using nvimtree. In this case when I open a new tab, the previous one's name becomes `NvimTree_number`. What I'd want to happen instead is having it be the name of the last selected buffer in the tab that passed the filter.

### Additional Information

...

**Comment by GabMus** [comment]
Thanks for the quick reply. I'll try to hack it together, with some luck I'll manage to put together something.

Unfortunately I'm not very familiar with lua or (n)vim plugin development, but I can certainly try 🙂

**Comment by akinsho** [comment]
Closed by #400 thanks again for the PR
